src/main/fc/config.c

```diff
@@ -513,6 +527,15 @@ void validateAndFixGyroConfig(void)
     }
 #endif
 
+    // Fix gyro filter settings to handle cases where an older configurator was used that
+    // allowed higher cutoff limits from previous firmware versions.
+    adjustFilterLimit(&gyroConfigMutable()->gyro_lowpass_hz);
+    adjustFilterLimit(&gyroConfigMutable()->gyro_lowpass2_hz);
+    adjustFilterLimit(&gyroConfigMutable()->gyro_soft_notch_hz_1);
```

Comment:
Erm, setting notch and cutoff frequency both to the same if they are both higher than `FILTER_FREQUENCY_MAX` will get them disabled in the code right below - I don't think this is what we are aiming for?

Reply:
The code considers each element individually and doesn't take into account logical relationships. If the user had both the center and cutoff above the limit then they will be implicitly disabled anyway by the nyquist limit code. Also I'm not sure we can default to any better values. Maybe the cutoffs should reset to 0 if out of range - but that will result in the notch being disabled as well.

Reply:
I think this relying on interaction of what these two parts of the code do to result in the desired outcome is not ideal. If we want a filter to be disabled if both frequencies are above the limit, then we should do so directly. Alternatively we could set the cutoff to some value below the limit, to make the transition from 'only center is above the limit' to 'both are above the limit' more gradual.